src/gateway/mod.ts

```diff
@@ -418,7 +418,7 @@ export class Gateway extends HarmonyEventEmitter<GatewayTypedEvents> {
     this.websocket.onerror = this.onerror.bind(this) as any
   }
 
-  close(code: number = 1000, reason?: string): void {
+  closeGateway(code: number = 1000, reason?: string): void {
```

Comment:
Not much happy with this rename.

Reply:
me neither but idk what else to call it because of event having close already :/

Reply:
And EventEmitter's close would be confusing too...

Reply:
that is true, but I feel like it is better to stay updated now than when it breaks and we possibly need to rewrite more code

Reply:
What's the exact error you get without renaming this?

Reply:
```ts
error: TS2416 [ERROR]: Property 'close' in type 'Gateway' is not assignable to the same property in base type 'HarmonyEventEmitter<GatewayTypedEvents>'.
  Type '(code?: number, reason?: string | undefined) => void' is not assignable to type '<K extends GatewayEvents | "connect" | "ping" | "resume" | "reconnectRequired" | "close" | "error" | "sentIdentify" | "sentResume" | "reconnecting" 
| "init">(eventName?: K | undefined) => Promise<...>'.
    Types of parameters 'code' and 'eventName' are incompatible.
      Type 'K | undefined' is not assignable to type 'number | undefined'.
        Type 'K' is not assignable to type 'number | undefined'.
          Type 'GatewayEvents | "connect" | "ping" | "resume" | "reconnectRequired" | "close" | "error" | "sentIdentify" | "sentResume" | "reconnecting" | "init"' is not assignable to type 'number | undefined'.
            Type 'GatewayEvents.Ready' is not assignable to type 'number | undefined'.
              Type 'K' is not assignable to type 'number'.
                Type 'string' is not assignable to type 'number'.
                  Type 'import("file:///C:/Users/User/Desktop/coding/projects/harmony_deps/src/types/gateway.ts").GatewayEvents' is not assignable to type 'number'.
  close(code: number = 1000, reason?: string): void {
  ~~~~~
    at file:///C:/Users/User/Desktop/coding/projects/harmony_deps/src/gateway/mod.ts:421:3

Reply:
Changing `close` to this helped.. but any types.
```ts
  close(...args: unknown[]): any {
    const code = args[0]
    const reason = args[1] ?? ''
    this.debug(
      `Closing with code ${code}${reason !== '' ? ` and reason ${reason}` : ''}`
    )
    return this.websocket?.close(code as number, reason as string)
  }
```

Reply:
Actually `// @ts-expect-error` would be fine here.

Reply:
idk if that is a good sollution either. Than I rather have a change in name rather than ignoring ts and using type `any`

Reply:
I think it would be better to use ts-expect-error here instead of changing name and creating more confusion. It is also a breaking change if name is changed.

Reply:
well, it is for v2 so like we can do it

Reply:
That's not the complete point. The name change is not much reasonable to me. @Helloyunho opinions?

Reply:
I think just changing the name is better. Since I don't want to ignore typescript errors.

Reply:
Ok then

Reply:
not making all things duplicated is better to recognize too.